doc/rfcs/001-allow-guests-to-open-multiple-remote-buffers.md

```diff
@@ -0,0 +1,47 @@
+# Allow guests to view multiple remote buffers
+
+## Status
+
+This is a proposal and is not yet implemented.
+
+## Summary
+
+Allow a separate guest editor for each open buffer in the host's workspace.
+
+## Motivation
+
+Currently, guests can only view host buffers through a specialized "portal editor" that automatically switches its contents to track the active editor of the host. This automatic switching can interrupt guests in the middle of typing. It also makes it impossible for guests to edit any buffer other than the one the host is actively editing.
+
+Ideally, guests would be able to edit any file in the host's current project, but that will require replication of the file-system which is beyond the scope of this proposal. We can get much closer to that ideal by giving guests access to any open buffer in the host's workspace. This can be implemented without substantial changes to the current replication architecture.
+
+## Explanation
+
+### Remote buffers
+
+Just as guests can open editors for local buffers corresponding to files on their local file system, they can also open editors for *remote buffers* corresponding to buffers in any remote workspace that they have joined.
+
+### Following
+
+When you first join a portal as a guest, you're automatically *following* the host. This means that you jump to the host's current position and continue to follow them as they move between buffers.
+
+In order to jump to the host's current position, an editor is automatically added to your workspace for their active buffer when you first join their workspace. If you continue to follow the host, any time they switch to a new buffer, a new editor for that remote buffer is automatically added to your workspace and focused. Existing editors for previous remote buffers are not automatically closed when this switch occurs.
+
+If you stop following the host, either manually or automatically due to them navigating away while you are still typing, then when they change their active editor your workspace is unaffected. If you start following the host again then a new editor may be added to your workspace based on their current location so that you can continue to follow them.
```

Comment:
> If you stop following the host, either manually or automatically due to them navigating away while you are still typing, then when they change their active editor your workspace is unaffected

So essentially what this means is that tabs are opened automatically only when you are following. If you are not, we will not open tabs for new buffers, but you can still access them via the file finder. Am I interpreting this correctly? If so, I really like this behavior! Do you think we could clarify such aspect a little more in this paragraph?

Reply:
Yeah, that was my intent. I can clarify this.